pkg/deployments/src/artifact.ts

```diff
@@ -8,12 +8,15 @@ import Task from './task';
 /**
  * Reads each of the task's build-info files and extract the ABI and bytecode for the matching contract.
  */
-export function extractArtifact(task: Task): void {
+export function extractArtifact(task: Task, contractName?: string): void {
   const buildInfoDirectory = path.resolve(task.dir(), 'build-info');
   if (existsSync(buildInfoDirectory) && statSync(buildInfoDirectory).isDirectory()) {
     for (const buildInfoFileName of readdirSync(buildInfoDirectory)) {
-      const contractName = path.parse(buildInfoFileName).name;
-      const artifact = extractContractArtifact(task, contractName);
+      if (contractName == undefined) {
+        contractName = path.parse(buildInfoFileName).name;
+      }
+
+      const artifact = extractContractArtifact(task, contractName, buildInfoFileName);
       writeContractArtifact(task, contractName, artifact);
     }
```

Comment:
Here we'd looking for a particular contract in all of the buildinfo files.

`extractContractArtifact` throws if it can't find the artifact for the given contract name so `yarn hardhat extract-artifacts` throws as soon as we hit a task which has a build-info file which doesn't contain the all the contracts which the other build-info files are named after (i.e. they come from different packages).

This means that we can't run this task for `2021-04-18-vault` as the Vault (`vault`) and BalancerHelpers (`standalone-utils`) breaks this requirement.

Reply:
Ah, true.

What if we made all tasks have a single build info?

Reply:
Eh, that would mean that we need to separate out tasks for `LiquidityGaugeV5` and `LiquidityGaugeFactory` (as the gauge is vyper so has a separate build-info) which seems ugly to me.

Reply:
I see. But surely having duplicated build infos is not a good idea either?